src/background/bgPopupHandler.ts

```diff
@@ -80,3 +80,34 @@ export const openPopup = async (
 export const removePopup = (id: string) => delete popups[id];
 
 export const getPopup = (id: string): IPopupConfigNoActions => popups[id];
+
+export const setSessionExpires = async (expires: number) => {
+  // @ts-expect-error session storage is not defined
+  await browser.storage.session.set({ sessionExpires: expires });
+};
+
+export const getSession = async () => {
+  try {
+    // @ts-expect-error session storage is not defined
+    const { sessionExpires } = await browser.storage.session.get('sessionExpires');
+    if (sessionExpires < Date.now() || !sessionExpires) {
+      // @ts-expect-error session storage is not defined
+      await browser.storage.session.remove('sessionKey');
+      return null;
+    }
+
+    // @ts-expect-error session storage is not defined
+    const { sessionKey } = await browser.storage.session.get('sessionKey');
+    if (sessionKey) {
+      return {
+        key: Buffer.from(sessionKey.key).toString('base64'),
+        salt: Buffer.from(sessionKey.salt).toString('base64'),
+        iv: Buffer.from(sessionKey.iv).toString('base64'),
+      };
+    }
+
+    return null;
+  } catch (error) {
+    return null;
+  }
```

Comment:
Possible different way of returning null. Ignore if you don't like it.

```suggestion
  } catch (error) {}
  
  return null;
```